composer/utils/collect_env.py

```diff
@@ -0,0 +1,391 @@
+    from torch.cuda import get_device_name as accel_device_name
+    from torch.cuda import is_available as cuda_available
+    TORCH_AVAILABLE = True
+except (ImportError,):
+    TORCH_AVAILABLE = False
+
+# Check if Composer is installed
+try:
+    import composer
+    from composer.utils import dist
+    COMPOSER_AVAILABLE = True
+except (ImportError,):
+    COMPOSER_AVAILABLE = False
+
+# Check if we're running in a notebook
+try:
+    __IPYTHON__  #type: ignore
+    IPYTHON_AVAILABLE = True
+except (NameError,):
+    IPYTHON_AVAILABLE = False
+
+if IPYTHON_AVAILABLE:
+    get_ipython = import_object('IPython:get_ipython')
+    nb = get_ipython()
+
+# Place to keep track of the original excepthook
+_orig_excepthook = None
+
+
+# Same convention as Torch collect_env, create a namedtuple to track collected fields
+class ComposerEnv(NamedTuple):
+    composer_version: str
+    node_world_size: int
+    host_processor_model_name: str
+    host_processor_core_count: int
+    local_world_size: int
+    accelerator_model_name: str
+    cuda_device_count: int
+
+
+# Helper functions to get Composer environment information
+def get_composer_version() -> str:
+    """Query the Composer version."""
+    return str(composer.__version__)
+
+
+def get_host_processor_name() -> str:
+    """Query the host processor name."""
+    cpu_info = cpuinfo.get_cpu_info()
+    return str(cpu_info.get('brand_raw', "CPU"))
+
+
+def get_host_processor_cores() -> int:
+    """Determine the number of physical host processor cores."""
+    return psutil.cpu_count(logical=False)
+
+
+def get_node_world_size() -> int:
+    """Query the number of nodes."""
+    return int(dist.get_world_size() / dist.get_local_world_size())
+
+
+def get_accel_model_name() -> str:
+    """Query the accelerator name."""
+    return accel_device_name(None) if cuda_available() else "N/A"
+
+
+def get_local_world_size() -> int:
+    """Determine the number of accelerators per node."""
+    return dist.get_local_world_size() if cuda_available() else 0
+
+
+def get_cuda_device_count() -> int:
+    """Get the number of CUDA devices on the system."""
+    return cuda_device_count() if TORCH_AVAILABLE else 0
+
+
+# Exception message and environment report
+COMPOSER_OPEN_ISSUE_URL = "https://github.com/mosaicml/composer/issues/new/choose"
+
+
+def _exc_report(exc_type) -> None:
+    """Produces exception report (exception message + environment report)
+
+    Args:
+        exc_type (Exception): Type of exception.
+    """
+    EXCEPTION_MSG = f"Bugs can be reported at: {COMPOSER_OPEN_ISSUE_URL}\n" + \
+                    f"Please include details on how to reproduce the issue and attach the following environment report:\n"
+
+    # Don't print exception report for KeyboardInterrupt
+    if not issubclass(exc_type, KeyboardInterrupt):
+        if issubclass(exc_type, AssertionError):
+            EXCEPTION_SEV_MSG = f"Smells like a Composer bug.\n"
```

Comment:
Asserts are for invariants, so they are always bugs if they surface (even if it means it should be a ValueError instead of an assert)

```suggestion
            EXCEPTION_SEV_MSG = f"This is a Composer bug. Please submit a bug report.\n"
```